File: 🟢 Easy/1684 - Count the Number of Consistent Strings/_1684CounttheNumberofConsistentStrings.cpp
class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        vector<bool> contains(26);
        int count = 0;

        for (int i = 0; i < allowed.length(); i++) {
            contains[allowed.at(i) - 'a'] = true;
        }

        bool notAllowed = false;

        for (auto& word : words) {
            notAllowed = false;

            for (int i = 0; i < word.length(); i++) {
                if (contains[word.at(i) - 'a'] == false) {
                    notAllowed = true;
                    break;
                }
            }

            if (!notAllowed) {
                count++;
            }
        }

        return count;
    }
};

File: 🟢 Easy/1684 - Count the Number of Consistent Strings/_1684CounttheNumberofConsistentStrings.js
/**
 * @param {string} allowed
 * @param {string[]} words
 * @return {number}
 */
var countConsistentStrings = function (allowed, words) {
    let contains = new Array(26).fill(false);
    let count = 0;

    for (let i = 0; i < allowed.length; i++) {
        contains[allowed.charCodeAt(i) - '97'] = true;
    }

    let notAllowed = false;

    for (let word of words) {
        notAllowed = false;

        for (let i = 0; i < word.length; i++) {
            if (contains[word.charCodeAt(i) - '97'] == false) {
                notAllowed = true;
                break;
            }
        }

        if (!notAllowed) {
            count++;
        }
    }

    return count;
};